exporters/memory/include/opentelemetry/exporters/memory/in_memory_span_exporter.h

```diff
@@ -0,0 +1,74 @@
+#pragma once
+#include "opentelemetry/exporters/memory/in_memory_span_data.h"
+#include "opentelemetry/sdk/trace/exporter.h"
+
+const size_t MAX_BUFFER_SIZE = 100;
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace exporter
+{
+namespace memory
+{
+/**
+ * A in memory exporter that switches a flag once a valid recordable was received
+ * and keeps track of all received spans in memory.
+ */
+class InMemorySpanExporter final : public opentelemetry::sdk::trace::SpanExporter
+{
+public:
+  /**
+   * @param buffer_size an optional value that sets the size of the InMemorySpanData
+   */
+  InMemorySpanExporter(size_t buffer_size = MAX_BUFFER_SIZE)
+      : data_(new opentelemetry::exporter::memory::InMemorySpanData(buffer_size))
+  {}
+
+  /**
+   * @return Returns a unique pointer to an empty recordable object
+   */
+  std::unique_ptr<sdk::trace::Recordable> MakeRecordable() noexcept override
+  {
+    return std::unique_ptr<sdk::trace::Recordable>(new sdk::trace::SpanData());
+  }
+
+  /**
+   * @param recordables a required span containing unique pointers to the data
+   * to add to the InMemorySpanData
+   * @return Returns the result of the operation
+   */
+  sdk::trace::ExportResult Export(
+      const nostd::span<std::unique_ptr<sdk::trace::Recordable>> &recordables) noexcept override
+  {
+    for (auto &recordable : recordables)
+    {
+      auto span = std::unique_ptr<sdk::trace::SpanData>(
+          static_cast<sdk::trace::SpanData *>(recordable.release()));
+      if (span != nullptr)
+      {
+        data_.get()->Add(std::move(span));
+      }
+    }
+
+    return sdk::trace::ExportResult::kSuccess;
+  }
+
+  /**
+   * @param timeout an optional value containing the timeout of the exporter
```

Comment:
Let's add a short note that this is not supported for this exporter.

Reply:
Added